src/api/wallet/composeProvider.ts

```diff
@@ -73,18 +75,13 @@ const wrapInTimeout = (middleware: JsonRpcMiddleware, timeout = DEFAULT_TX_APPRO
 
     let timeoutId: NodeJS.Timeout | null = null
 
-    timeoutId = setTimeout(function askOnTimeout() {
-      if (confirm('Stop waiting for tx approval?')) {
+    timeoutId = setTimeout(async function askOnTimeout() {
+      if (await openWaitForTxApprovalModal(timeout)) {
```

Comment:
This has the undesired effect that on multiple tx, you get a lot of confirmations

Can we do sth in the direction of keeping a count of the number of transactions waiting.

- When you send transactions to the wallet, you increment
- When they resolve or reject, you decrement
- When the timer fires, we ask for confirmation to stop waiting, and if the user cancels the waiting, we decrement

Only when our count is 0, we can stop waiting.

Reply:
Thankfully we can't send multiple trade txs, but yes there are cases.
Need a count somewhere

Reply:
Not sure how to go about implementing this
If we use one modal for all pending txs that timed out:
- Do we just not close it if the count > 0
- Do we close and reopen
- Do we allow to stop waiting on all txs together
- Do we update the modal in real time if another tx becomes late
- Maybe throttle all tx approvals, but then they can potentially come out of order, depending on wallet

I don't see it being a good UX either way

@W3stside chime in please

Reply:
oof. uhhhh shit. idk what to input here... This sounds super edgecase-y and the amount of work seems... heavy.

But a global state var keeping track of current ongoing txs is a start, then we have 1 modal? and wait there? add any new incoming @Velenir